lib/state/simperium/middleware.ts

```diff
@@ -246,12 +246,14 @@ export const initSimperium = (
       case 'FILTER_NOTES':
       case 'OPEN_NOTE':
       case 'SELECT_NOTE': {
+        const state = getState();
         const noteId =
           action.noteId ??
           action.meta?.nextNoteToOpen ??
           getState().ui.openedNote;
 
-        if (noteId) {
+        //  Preload the revisions when opening a note but only do it if no revisions are in memory
+        if (noteId && !state.data.noteRevisions.get(noteId)?.size) {
           setTimeout(() => {
             if (getState().ui.openedNote === noteId) {
```

Comment:
note that here we're using `getState()` because this code is run inside the callback and `prevState` and `nextState` will already be expired. this code runs 250ms in the future and we have to call `getState()` again to the get the up-to-date value